### PR TITLE
Extended OfflinePlayer API

### PR DESCRIPTION
**Pull request by YouHaveTrouble**
- Added some methods to OfflinePlayer class.
- Added convienience method for saving player file after modifications to the nbt data
- All methods fall back to getPlayer methods if the player is online

edit: apparently build failed, going to bed, gonna fix when I wake up
edit2: builds and runs for me locally

### CODE REVIEW

File: patches/server/0252-Extended-OfflinePlayer-API.patch [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+ }
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index 3230631551e530afdf0022a4c863cc57c0374df2..93febcd9cbb5d81474b61f931a63e22293729802 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -1925,6 +1925,26 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         return this.getHandle().getAbilities().walkingSpeed * 2f;
+     }
+ 
++    @Override
```
Comment (katekerllenevich) [inline patches/server/0252-Extended-OfflinePlayer-API.patch:234]: Purpur comments?

Comment (YouHaveTrouble) [reply patches/server/0252-Extended-OfflinePlayer-API.patch:234]: oh, forgot. Fixing now

**Review by katekerllenevich — APPROVED** [review]
lgtm, waiting for billy.

**Comment by YouHaveTrouble** [comment]
There's a small discrepency between the javadoc and impl, fixing in next commit

**Review by BillyGalbreath — APPROVED** [review]

**Pull request merged**